src/piqtree2/iqtree/_tree.py

```diff
@@ -134,12 +134,15 @@ def _process_tree_yaml(
     newick = tree_yaml["PhyloTree"]["newick"]
 
     tree = cogent3.make_tree(newick)
-
+    t2t_matrix = tree.tip_to_tip_distances()[0]
```

Comment:
This algorithm only checks whether two distance matrices for a tree are equal. It does not check their names (fails a new test I wrote). I'd recommend writing a new function `tree_equal` which checks both the names and branch lengths of two trees. This can be done as a recursive function.